### PR TITLE
feat(scala3): make specs2 run

### PR DESCRIPTION
**Pull request by mantasak**
### Description
Made specs2 build on Scala3.1.0

### Motivation
Needed to start supporting Scala3 in Wix.

### CODE REVIEW

File: src/java/io/bazel/rulesscala/specs2/FilteredSpecs2ClassRunner.scala [hunk 1]
```diff
@@ -28,15 +30,21 @@ class FilteredSpecs2ClassRunner(testClass: Class[_], testFilter: Pattern)
...
-//    descTree.toTree.bottomUp {
-//      (description: Description, children: Stream[Description]) =>
-//        children.filter(matchingFilter).foreach {
-//          child => description.addChild(child)
-//        }
-//        description
-//    }.rootLabel
-    descTree.getLabel
 
+    def bottomUp[A, B](t: Tree[A], f: ((A, Iterable[B]) => B)): Tree[B] = {
```
Comment (mantasak) [inline src/java/io/bazel/rulesscala/specs2/FilteredSpecs2ClassRunner.scala:34]: had to copy code and modify slightly to use generic Iterable to ensure it builds on both S2 and S3

File: examples/testing/multi_frameworks_toolchain/example/Specs2ExampleTest.scala [hunk 2]
```diff
@@ -3,9 +3,7 @@ package example
 import org.specs2.mutable.SpecWithJUnit
 
 class Specs2ExampleTest extends SpecWithJUnit {
-
-  "Example" in {
-    1 mustEqual 1
+  "Example" >> {
```
Comment (liucijus) [inline examples/testing/multi_frameworks_toolchain/example/Specs2ExampleTest.scala:6]: Can you explain such changes?

File: scala/private/phases/phase_compile.bzl [hunk 3]
```diff
@@ -106,7 +106,7 @@ def _phase_compile_default(ctx, p, _args = struct()):
         ctx,
         p,
         _args.srcjars if hasattr(_args, "srcjars") else depset(),
-        _args.buildijar if hasattr(_args, "buildijar") else True,
+        _args.buildijar if hasattr(_args, "buildijar") else False,
```
Comment (mantasak) [inline scala/private/phases/phase_compile.bzl:109]: this forces bazel to use full jar, otherwise compiling does not find classes. i think this may be related to how scala3 uses tasty to ensure strong type safety, but not sure why it needs the .class implementations... (tasty does exist on the ijar)

File: examples/testing/multi_frameworks_toolchain/example/Specs2ExampleTest.scala (hunk 2 above)
Comment (mantasak) [reply examples/testing/multi_frameworks_toolchain/example/Specs2ExampleTest.scala:6]: Yeah. I was writing comments, but I will here.

In specs2 v5 `should` is defined both as expect matcher (like must) and as dsl. Unfortunately without specifying extended trait (ExtendedBlockDsl) it does not know which one to use and compile fails. I could not extend ExtendedBlockDsl, because in v4 of specs2 it does not exist.

Also, mustEqual is not present in v5, and I substituted it for another matcher.

Comment (mantasak) [reply examples/testing/multi_frameworks_toolchain/example/Specs2ExampleTest.scala:6]: Regarding `in`, I had to extend the same DSL builder in some of the test classes, so I decided to align them with `should` replacements and just use >> which is cross-compatible. I could not debug exactly why, because the entire targets don't build on my intellij due to other errors...

File: scala/private/phases/phase_compile.bzl (hunk 3 above)
Comment (liucijus) [inline scala/private/phases/phase_compile.bzl:109]: I think this change can be contributed as a separate PR to rules scala, just do a conditional according scala version. Example:
```
load("@io_bazel_rules_scala_config//:config.bzl", "SCALA_MAJOR_VERSION", "SCALA_VERSION")

buildijar_default_value = True if SCALA_VERSION.startswith("2.") else False
```
